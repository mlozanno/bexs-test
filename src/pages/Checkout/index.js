import React from 'react';

const Checkout = () => <div data-testid="checkout-page" />;

export default Checkout;
