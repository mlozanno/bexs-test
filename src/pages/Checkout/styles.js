import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { pxToRem, colors, fontSize } from '~/styles/theme';

import CreditCard from '~/components/CreditCard';

import titleIcon from '~/assets/new-card-icon.svg';

export const StyledCardContainer = styled.section`
	@media (min-width: 1024px) {
		display: flex;
		height: 100vh;
	}
`;

export const CardContent = styled.aside`
	position: relative;
	display: flex;
	flex-direction: column;
	align-items: center;
	margin-bottom: ${pxToRem(100)};
	padding: ${pxToRem(40, 40, 114)};
	color: ${colors.light};
	font-size: ${fontSize.small};
	background-color: ${colors.primary};

	@media (min-width: 1024px) {
		align-items: flex-start;
		width: ${pxToRem(350)};
		height: 100%;
		margin-bottom: 0;
		padding: ${pxToRem(50, 0, 0, 64)};
	}
`;

export const BackLink = styled(Link)`
	position: absolute;
	top: ${pxToRem(35)};
	left: ${pxToRem(5)};

	> span {
		display: none;
	}

	@media (min-width: 1024px) {
		position: static;
		display: flex;
		align-items: center;

		> span {
			display: inline-block;
			margin-left: ${pxToRem(14)};
			color: ${colors.light};
			font-size: ${fontSize.small};
		}
	}
`;

export const NewCardTitle = styled.h2`
	position: relative;
	max-width: ${pxToRem(220)};
	margin-top: ${pxToRem(30)};
	padding-left: ${pxToRem(55)};
	color: inherit;
	font-size: ${pxToRem(16)};

	@media (min-width: 1024px) {
		margin: ${pxToRem(50, 0, 30)};
	}

	&::before {
		position: absolute;
		top: calc(50% - ${pxToRem(20)});
		left: 0;
		width: ${pxToRem(40)};
		height: ${pxToRem(40)};
		background: url(${titleIcon}) no-repeat;
		background-size: contain;
		content: '';
	}
`;

export const StyledCreditCard = styled(CreditCard)`
	position: absolute;
	top: ${pxToRem(150)};

	@media (min-width: 1024px) {
		position: relative;
		top: auto;
	}
`;

export const FormContent = styled.div`
	background-color: ${colors.light};

	@media (min-width: 1024px) {
		padding: ${pxToRem(53, 64, 0, 168)};
	}
`;
