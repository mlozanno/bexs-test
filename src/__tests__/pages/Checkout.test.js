import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Checkout from '~/pages/Checkout';

describe('<Checkout /> page', () => {
	it('should be a function', () => {
		expect(typeof Checkout).toBe('function');
	});

	it('should be render component', () => {
		const { getByTestId } = render(<Checkout />);

		const checkout = getByTestId('checkout-page');

		expect(checkout).toMatchSnapshot();
	});
});
