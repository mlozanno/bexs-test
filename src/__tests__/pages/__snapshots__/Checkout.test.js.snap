// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> page should be render component 1`] = `
<section
  data-testid="checkout-page"
>
  <section
    class="sc-AxgMl fFPQpl"
  >
    <div
      class="sc-AxjAm bqZbHY"
      data-testid="credit-card"
    >
      <div
        class="sc-AxirZ hAbGUU"
      >
        <div
          class="sc-AxiKw iZbdSR"
        >
          <div
            class="sc-AxhCb iYtyOb"
          >
            <span
              data-field="card-number"
              data-testid="credit-card-number"
            >
              **** **** **** ****
            </span>
            <span
              data-field="card-user-name"
              data-testid="credit-card-user-name"
            >
              NOME DO TITULAR
            </span>
            <span
              data-field="card-expiration-date"
              data-testid="credit-card-expiration-date"
            >
              00/00
            </span>
          </div>
          <div
            class="sc-AxhUy dvkhgG"
          >
            <span
              data-field="card-cvv"
              data-testid="credit-card-cvv"
            >
              ***
            </span>
          </div>
        </div>
      </div>
      <h1>
        Flipped:
        true
      </h1>
      <button>
        Flippar
      </button>
      <button>
        Validar
      </button>
    </div>
  </section>
</section>
`;
